proto/api/v1/tenant.proto

```diff
@@ -182,3 +248,80 @@ message TenantServiceDeleteResponse {
   // Tenant is the tenant
   Tenant tenant = 1;
 }
+
+// TenantServiceInviteRequest is used to invite a member to a tenant
+message TenantServiceInviteRequest {
+  // Tenant is the uuid of the tenant
+  string tenant = 1 [(validate.rules).string.uuid = true];
+
+  // Role of this user in this tenant
+  TenantRole role = 3;
+}
+
+// TenantServiceInviteRequest is the response payload to a invite member request
+message TenantServiceInviteResponse {
+  // Inviter contains a secret which can be sent to a potential user
+  // can appended to the invitation endpoint at our api server like
+  // console.metalstack.cloud/invite/<secret>
+  TenantInvite invite = 1;
+}
+
+// TenantServiceInvitesListRequest is the request payload to a list invites request
+message TenantServiceInvitesListRequest {
+  // Tenant is the uuid of the tenant
+  string tenant = 1 [(validate.rules).string.uuid = true];
+}
+
+// TenantServiceInvitesListResponse is the response payload to a list invites request
+message TenantServiceInvitesListResponse {
+  // Invites not already accepted the invitation to this tenant
+  repeated TenantInvite invites = 1;
+}
+
+// TenantServiceInviteGetRequest is the request payload to get a invite
+message TenantServiceInviteGetRequest {
+  // Secret of the invite to list
+  string secret = 1;
+}
+
+// TenantServiceInviteGetResponse is the response payload to a get invite request
+message TenantServiceInviteGetResponse {
+  // Invite is the invite
+  TenantInvite invite = 1;
+}
+
+// TenantServiceRemoveMemberRequest is used to remove a member from a tenant
+message TenantServiceRemoveMemberRequest {
+  // Tenant is the uuid of the tenant
+  string tenant = 1 [(validate.rules).string.uuid = true];
```

Comment:
If we want to maintain old tenant ID naming convention, we cannot validate UUIDs here.

Reply:
removed this validation